### PR TITLE
Fix serialize() usage in permissionSet.save()

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Also add test for .save()

### CODE REVIEW

**Comment by dmitrizagidulin** [comment]
Tagging @dan-f / @nicola / @deiu for review.

File: src/permission-set.js [hunk 1]
```diff
@@ -719,11 +720,12 @@ class PermissionSet {
    * @method save
    * @param [aclUrl] {String} Optional URL to save the .ACL resource to. Defaults
    *   to its pre-set `aclUrl`, if not explicitly passed in.
+   * @param [contentType] {string} Optional content type to serialize it as
    * @throws {Error} Rejects with an error if it doesn't know where to save, or
    *   with any XHR errors that crop up.
    * @return {Promise<Request>}
    */
-  save (aclUrl, rdf, webClient) {
+  save (aclUrl, contentType, rdf, webClient) {
```
Comment (dmitrizagidulin) [inline src/permission-set.js]: Sure, good idea

File: src/permission-set.js [hunk 2]
```diff
@@ -736,7 +738,10 @@ class PermissionSet {
     if (!webClient) {
       return Promise.reject(new Error('Cannot save - no web client'))
     }
-    return webClient.put(aclUrl, this.serialize(rdf))
+    contentType = contentType || DEFAULT_CONTENT_TYPE
+    let graph = this.serialize(rdf, contentType)
+    return Promise.resolve()
```
Comment (elf-pavlik) [inline src/permission-set.js]: maybe
```js
return this.serializer(rdf)
  .then( (graph) => { return webClient.put(aclUrl, graph, contentType })
```

File: src/permission-set.js (hunk 1 above)
Comment (elf-pavlik) [inline src/permission-set.js]: actually if one can set all above except contentType in the constructor, what use cases you have for overwriting it in save?

File: src/permission-set.js [hunk 3]
```diff
@@ -736,7 +738,10 @@ class PermissionSet {
     if (!webClient) {
       return Promise.reject(new Error('Cannot save - no web client'))
     }
-    return webClient.put(aclUrl, this.serialize(rdf))
+    contentType = contentType || DEFAULT_CONTENT_TYPE
```
Comment (dan-f) [inline src/permission-set.js]: You can use es6 default params for this :)

File: src/permission-set.js (hunk 2 above)
Comment (dan-f) [inline src/permission-set.js]: Why are you returning `Promise.resolve()`? Shouldn't this just be `return webClient.put(aclUrl, graph, contentType)`

File: test/unit/permission-set-test.js [hunk 4]
```diff
@@ -259,3 +260,25 @@ test('PermissionSet init from untyped ACL test', function (t) {
...
+  let isContainer = false
+  let putSpy = sinon.spy()
+  let mockWebClient = {
+    put: putSpy
+  }
+  let ps = new PermissionSet(resourceUrl, aclUrl, isContainer,
+    { rdf, webClient: mockWebClient })
+  ps.save()
+    .then(() => {
+      t.ok(putSpy.calledWith(aclUrl),
```
Comment (dan-f) [inline test/unit/permission-set-test.js]: Aren't you also going to test that the correct graph was passed to the web client?

File: src/permission-set.js (hunk 2 above)
Comment (elf-pavlik) [inline src/permission-set.js]: this.serialize() returns a Promise, so currently it looks that a Promise gets passed to webClient.put as graph

**Comment by dmitrizagidulin** [comment]
@dan-f added tests to save() to make sure the right graph is being saved, etc. Edited to reflect feedback.

File: test/unit/permission-set-test.js [hunk 5]
```diff
@@ -216,7 +237,7 @@ test('PermissionSet equals test 4', function (t) {
 
 test('PermissionSet serialized & deserialized round trip test', function (t) {
   var ps = new PermissionSet(containerUrl, containerAclUrl,
-    PermissionSet.CONTAINER, { graph: parsedAclGraph, rdf: rdf })
+    PermissionSet.CONTAINER, { graph: parsedAclGraph, rdf })
```
Comment (nicola) [inline test/unit/permission-set-test.js]: do we have a preference for `rdf: rdf`?

File: src/permission-set.js [hunk 6]
```diff
@@ -717,26 +715,27 @@ class PermissionSet {
...
+   *   Defaults to its pre-set `aclUrl`, if not explicitly passed in.
+   * @param [options.contentType] {string} Optional content type to serialize as
    * @throws {Error} Rejects with an error if it doesn't know where to save, or
    *   with any XHR errors that crop up.
-   * @return {Promise<Request>}
+   * @return {Promise<SolidResponse>}
    */
-  save (aclUrl, rdf, webClient) {
-    aclUrl = aclUrl || this.aclUrl
+  save (options = {}) {
```
Comment (dan-f) [inline src/permission-set.js]: A neat trick is to squash the next couple of lines into the function signature like this, which I also find easier to read:

```js
save ({ aclUrl = this.aclUrl, contentType = DEFAULT_CONTENT_TYPE }) { ... }
```

File: src/permission-set.js [hunk 7]
```diff
@@ -746,20 +745,24 @@ class PermissionSet {
...
    */
-  serialize (contentType, rdf) {
-    contentType = contentType || 'text/turtle'
-    rdf = rdf || this.rdf
-    var graph = this.buildGraph(rdf)
-    var target = null
-    var base = null
-    return new Promise(function (resolve, reject) {
-      rdf.serialize(target, graph, base, contentType, function (err, result) {
+  serialize (options = {}) {
```
Comment (dan-f) [inline src/permission-set.js]: Same note here about the function signature as above.

File: src/permission-set.js (hunk 6 above)
Comment (dmitrizagidulin) [inline src/permission-set.js]: Whoa, sweet! I didn't realize that was an option! Changing it...

**Pull request merged**